### PR TITLE
translate header label and group name

### LINKED ISSUE #537: The group label not translate
in the source code,the group label has not translate.

i find that ,if i do not modify the source code ,i can not translate it !

image
image

### PR DESCRIPTION
**Pull request by polarloves**
Closes #537

### CODE REVIEW

File: src/render/PanelHeaderProvider.js [hunk 1]
```diff
@@ -79,10 +83,8 @@ export const PanelHeaderProvider = {
 
   getTypeLabel: (element) => {
     const concreteType = getConcreteType(element);
-
-    return concreteType
-      .replace(/(\B[A-Z])/g, ' $1')
-      .replace(/(\bNon Interrupting)/g, '($1)');
```
Comment (pinussilvestrus) [inline src/render/PanelHeaderProvider.js:85]: Why is this removed?

File: src/provider/bpmn/BpmnPropertiesProvider.js [hunk 2]
```diff
@@ -206,17 +206,23 @@ function getGroups(element) {
...
+        group.label=translate(group.label)
+      })
       return groups;
     };
   }
 
 }
 
-BpmnPropertiesProvider.$inject = [ 'propertiesPanel' ];
+BpmnPropertiesProvider.$inject = [ 'propertiesPanel','translate' ];
```
Comment (pinussilvestrus) [inline src/provider/bpmn/BpmnPropertiesProvider.js:228]:
```suggestion
BpmnPropertiesProvider.$inject = [ 'propertiesPanel', 'translate' ];
```

File: src/provider/bpmn/BpmnPropertiesProvider.js [hunk 3]
```diff
@@ -206,17 +206,23 @@ function getGroups(element) {
...
+    BpmnPropertiesProvider.prototype.translate=translate
     propertiesPanel.registerProvider(this);
   }
 
   getGroups(element) {
+    const translate=this.translate
     return (groups) => {
       groups = groups.concat(getGroups(element));
+      // translate group name
+      groups.map(function(group){
```
Comment (pinussilvestrus) [inline src/provider/bpmn/BpmnPropertiesProvider.js:219]: Minor thing: Why do we use `map` instead of `forEach`? We are not mapping anything here into a new array.

File: src/provider/bpmn/BpmnPropertiesProvider.js [hunk 4]
```diff
@@ -206,17 +206,23 @@ function getGroups(element) {
 
 export default class BpmnPropertiesProvider {
 
-  constructor(propertiesPanel) {
+  constructor(propertiesPanel,translate) {
+    BpmnPropertiesProvider.prototype.translate=translate
```
Comment (pinussilvestrus) [inline src/provider/bpmn/BpmnPropertiesProvider.js:210]:
```suggestion
    this._translate = translate;
```

File: src/provider/bpmn/BpmnPropertiesProvider.js [hunk 5]
```diff
@@ -206,17 +206,23 @@ function getGroups(element) {
 
 export default class BpmnPropertiesProvider {
 
-  constructor(propertiesPanel) {
+  constructor(propertiesPanel,translate) {
```
Comment (pinussilvestrus) [inline src/provider/bpmn/BpmnPropertiesProvider.js:209]:
```suggestion
  constructor(propertiesPanel, translate) {
```

File: src/provider/bpmn/BpmnPropertiesProvider.js [hunk 6]
```diff
@@ -206,17 +206,23 @@ function getGroups(element) {
...
     propertiesPanel.registerProvider(this);
   }
 
   getGroups(element) {
+    const translate=this.translate
     return (groups) => {
       groups = groups.concat(getGroups(element));
+      // translate group name
+      groups.map(function(group){
+        group.label=translate(group.label)
```
Comment (pinussilvestrus) [inline src/provider/bpmn/BpmnPropertiesProvider.js:220]:
```suggestion
        group.label = translate(group.label);
```

**Review by pinussilvestrus — COMMENTED** [review]
First thanks for opening this pull request 👍  I left some comments we should consider.

Next to those, there are two things:
* Let's update the commit message to keep our history clean, e.g.

```text
feat(bpmn): translate group labels
feat(render): translate header type
```

Also make sure that one of these commits are referencing the closing issue

```text
Closes #537 
```

* Can you think about any test case to verify it works as expected?

**Comment by polarloves** [comment]
thanks !  so many useful suggest , i will close this pull request and start another!
